### DISCUSSION TITLE
Dreamhost : Can only update @ record?

### DISCUSSION THREAD

**Discussion by enphor** [qdm12/ddns-updater#256] [Q&A] [answered]
Hello, I currently am using v2.4.1 with Unraid in order to update my Dreamhost DNS via the API. I tried to include the 'host:' directive but received an error when restarting the container that only updating the '@' A record is supported. Am I doing something wrong or is this a limitation? I was previously using pfsense with their ddns updater tool with Dreamhost via API and it was able to update any A record. Thank you for your help!

**Comment by qdm12** [comment, marked as answer]
Hey there! Not sure why, I can't remember 😵‍💫 

I pushed an image (only amd64) `qmcgaw/ddns-updater:256` to support different hosts, can you try it please?
Also let me know if it works for wildcard `*` hosts, that I'm 100% not sure it will work...

Thanks!

**Reply by enphor** [reply]
Hello, thank you for such a quick response!

If I did everything right, I am running your 256 image using your docker-compose script. It won't let me use the host: directive in the config.json with Dreamhost.

```
ddns-updater  | Running version latest built on 2021-10-15T14:43:47Z (commit 14cd7b0)
ddns-updater  |
ddns-updater  | 🔧 Need help? https://github.com/qdm12/ddns-updater/discussions/new
ddns-updater  | 🐛 Bug? https://github.com/qdm12/ddns-updater/issues/new
ddns-updater  | ✨ New feature? https://github.com/qdm12/ddns-updater/issues/new
ddns-updater  | ☕ Discussion? https://github.com/qdm12/ddns-updater/discussions/new
ddns-updater  | 💻 Email? quentin.mcgaw@gmail.com
ddns-updater  | 💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
ddns-updater  | 2021/10/21 14:38:13 ERROR host can only be "@"
ddns-updater  | 2021/10/21 14:38:13 INFO Shutdown successful
``` 

Substituted my domain with "domain.net" for privacy reasons in the below config.json output.

```json
{
     "settings": [
         {
             "provider": "dreamhost",
             "domain": "domain.net",
             "host": "apple",
             "key": "********"
         }
     ]
}
```

**Reply by qdm12** [reply]
I don't think this is image `:256`. The logs shows `latest built on 2021-10-15T14:43:47Z (commit 14cd7b0)`, make sure you updated the image tag name.

**Reply by enphor** [reply]
Hello you were correct, sorry I'm a docker newb. I only use it when I need to get something working, then I re-learn everything and promptly forget until next time!

Your qmcgaw/ddns-updater:256 image works great! I was able to use the host: directive and added a host named apple which did not exist in dns for my domain. The app identified that it did not exist, created it, and updated the IP correctly. Thanks so much! 💯 ! I threw a small sponsorship your way. 😃 

Quick question , will you be able to add this update to Unraid Community Apps? 

```
2021/10/24 14:33:14 INFO reading JSON config from file /updater/data/config.json
2021/10/24 14:33:14 INFO Found single setting to update record
2021/10/24 14:33:14 INFO Reading history from database: domain domain.net host apple
2021/10/24 14:33:14 INFO backup: disabled
2021/10/24 14:33:14 INFO healthcheck server: listening on 127.0.0.1:9999
2021/10/24 14:33:14 INFO http server: listening on :8000
2021/10/24 14:33:15 WARN cannot DNS resolve apple.domain.net after 5 tries: lookup apple.domain.net on 10.0.25.1:53: no such host
2021/10/24 14:33:15 INFO IP address of apple.domain.net is <nil> and your IP address is 123.x
2021/10/24 14:33:15 INFO Updating record [domain: domain.net | host: apple | provider: dreamhost | ip: ipv4 or ipv6] to use 123.x
```

**Reply by qdm12** [reply]
Awesome! I merged the changes in ecda99a2176bde47e8716255b1b530490e95f523 in the latest image (give it 5min to build). 

I'm not too familiar with Unraid, but I think pulling the latest image `qmcgaw/ddns-updater` and restarting the container will add the updates. I don't think you need to change the template for that.

The template is at https://github.com/DiamondPrecisionComputing/unraid-templates/blob/main/templates/ddns-updater.xml. If you see something feel free to edit the file in your browser to do a Pull request, @DiamondPrecisionComputing is usually fast at answering 👍

**Reply by DiamondPrecisionComputing** [reply]
As long as the template is still pointed to the latest image then it will automatically pickup the changes. Or you can do a "force update". Either way should work.

**Reply by enphor** [reply]
Excellent, yes Unraid pulls from qmcgaw/ddns-updater and works fine. Thanks again
